File: frontend/src/app/core/services/admin-organization.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpParams } from '@angular/common/http';
import { ApiService } from './api.service';
import { AdminOrg, AdminOrgDetail, Page } from '../../models/admin-org.model';
import { AdminAutomationSummary, AdminMailbox } from '../../models/admin.model';

/**
 * Platform-staff API client for managing organizations (tenants):
 * listing, detail lookup, suspend/activate, ownership transfer and deletion.
 */
@Injectable({ providedIn: 'root' })
export class AdminOrganizationService {
  private readonly api = inject(ApiService);
  private readonly base = '/admin/organizations';

  list(search = '', personal?: boolean, page = 0, size = 20) {
    let params = new HttpParams()
      .set('search', search)
      .set('page', page)
      .set('size', size);
    if (personal !== undefined) {
      params = params.set('personal', personal);
    }
    return this.api.get<Page<AdminOrg>>(this.base, { params });
  }

  get(id: string) {
    return this.api.get<AdminOrgDetail>(`${this.base}/${id}`);
  }

  /** Automations owned by the org (capped at 200, newest first — no pagination). */
  getAutomations(id: string) {
    return this.api.get<AdminAutomationSummary[]>(`${this.base}/${id}/automations`);
  }

  /** Mailboxes (email accounts) owned by the org. No credentials are exposed. */
  getMailboxes(id: string) {
    return this.api.get<AdminMailbox[]>(`${this.base}/${id}/mailboxes`);
  }

  suspend(id: string, reason?: string) {
    return this.api.post<AdminOrgDetail>(`${this.base}/${id}/suspend`, { reason: reason ?? null });
  }

  activate(id: string) {
    return this.api.post<AdminOrgDetail>(`${this.base}/${id}/activate`, {});
  }

  transferOwnership(id: string, newOwnerUserId: string) {
    return this.api.post<AdminOrgDetail>(`${this.base}/${id}/transfer-ownership`, { newOwnerUserId });
  }

  remove(id: string) {
    return this.api.delete<void>(`${this.base}/${id}`);
  }
}

File: backend/src/main/java/com/postwerk/service/AdminOrganizationService.java
package com.postwerk.service;

import com.postwerk.dto.admin.AdminAutomationSummaryResponse;
import com.postwerk.dto.admin.AdminMailboxResponse;
import com.postwerk.dto.admin.AdminOrgDetailResponse;
import com.postwerk.dto.admin.AdminOrgResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

/**
 * Platform-staff oversight of customer organizations (tenants): listing, detail, ownership transfer
 * and soft-deletion. Separate from the tenant-facing {@link OrganizationService} (which is scoped to
 * the caller's own org) — this operates across ALL organizations and is gated by {@code ORG_*}
 * staff permissions.
 *
 * @since 1.0
 */
public interface AdminOrganizationService {

    Page<AdminOrgResponse> listOrganizations(String search, Boolean personal, Pageable pageable);

    AdminOrgDetailResponse getOrganization(UUID orgId);

    /** Automations owned by the organization (admin org-detail "Automations" tab). Mirrors {@code automationCount}'s ownership. */
    List<AdminAutomationSummaryResponse> getOrganizationAutomations(UUID orgId);

    /** Email accounts owned by the organization (admin org-detail "Mailboxes" tab). Mirrors {@code mailboxCount}'s ownership. */
    List<AdminMailboxResponse> getOrganizationMailboxes(UUID orgId);

    /** Transfers ownership to an existing member (new owner → OWNER, previous owner → ADMIN). */
    AdminOrgDetailResponse transferOwnership(UUID orgId, UUID newOwnerUserId);

    /**
     * Suspends a team organization: members are blocked from accessing it (their {@code X-Org-Id}
     * requests are rejected) until reactivated. Personal organizations cannot be suspended.
     *
     * @param reason optional staff note recorded for support/audit (may be {@code null}/blank)
     */
    AdminOrgDetailResponse suspendOrganization(UUID orgId, String reason);

    /** Lifts a suspension, restoring tenant access. No-op if the organization is already active. */
    AdminOrgDetailResponse activateOrganization(UUID orgId);

    /** Soft-deletes a team organization. Personal organizations cannot be deleted. */
    void deleteOrganization(UUID orgId);
}
